proposals/contour.adoc

```diff
@@ -0,0 +1,88 @@
+== Contour Proposal
+
+*Name of project:* Contour
+
+*Description:* Contour is an open source Kubernetes ingress controller providing the control plane for the Envoy edge and service proxy. Contour supports dynamic configuration updates and multi-team ingress delegation out of the box, while maintaining a lightweight profile.
+
+=== Alignment with CNCF - Why does CNCF need an ingress controller?
+
+The CNCF has an impressive portfolio of projects that can be leveraged to build and run complex distributed systems; our team believes Contour to be a great fit for the CNCF. Contour's core mission aligns well with Kubernetes and Envoy and the container and networking ecosystem. The CNCF's mission is to “create and drive the adoption of a new computing paradigm that is optimized for modern distributed systems environments capable of scaling to tens of thousands of self-healing multi-tenant nodes.” Modern distributed systems rely on networking and connectivity. As a result, ingress controllers for the compute platform, Kubernetes, are an essential piece of this architecture. Contour is a logical complement to Envoy in this architecture, making it easier to consume Envoy in a cloud native multi-team environment.
+
+=== Contour Overview
+
+==== Features
+
+ * Envoy Inside - Contour is built as the control plane for Envoy, the high performance L7 proxy and load balancer
+ * Flexible Architecture - Contour can be deployed as either a Kubernetes Deployment or DaemonSet
+ * Multi-team support - Safely support ingress in multi-team Kubernetes clusters
+ * TLS Certificate Delegation - Administrators can delegate wildcard certificate access securely
+ * Support for outputting HTTP request logs in a configurable structured JSON format
+ * One service per route can be nominated as a read-only mirror. The mirror service will receive a copy of the read traffic sent to any non-mirror service
+
+To learn more about Contour's features, read and view the following resources:
+
+ * https://projectcontour.io/announcing-contour-1.0/[Intro to Contour]
+ * https://projectcontour.io/routing-traffic-to-applications-in-kubernetes-with-contour/[Routing Traffic to Applications in Kubernetes with Contour]
+ * https://projectcontour.io/httpproxy-in-action/[HTTPProxy in Action]
+ * https://projectcontour.io/resources/[Contour Resources]
+
+=== Project Timeline and Snapshot
+ * Heptio launched and open sourced Contour in October 2017 as an ingress controller for Kubernetes based on Envoy
+ * Contour 1.0 ships in November 2019, signaling to the community a commitment to a stable set of APIs 
+ 
+== Production Users
+ * TBD
+
+== In-Flight Features
+
+The Contour team is currently working on the following feature improvements:
+ * Using Contour to redirect requests to locations other than Pods, rewriting the ExternalName in the host header for proxied requests
+ * Migration tooling for IngressRoute to HTTPProxy
+
```

Comment:
Should [service-apis](https://github.com/kubernetes-sigs/service-apis) support be added here?

/cc @youngnick.

Reply:
added. thanks!